src/content/en/updates/2017/09/abortable-fetch.md

```diff
@@ -0,0 +1,292 @@
+      console.log(chapters);
+    });
+
+In this case, calling `controller.abort()` will either abort whichever fetches are in-progress.
+
+## The future
+
+### Other browsers
+
+Firefox did a great job to ship this first. Their engineers implemented from the [test
+suite](https://github.com/w3c/web-platform-tests/tree/master/fetch/api/abort) while the spec was
+being written. For other browsers, here are the tickets to follow:
+
+* [Edge](https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/13009916/) – closed as
+  "fixed", but it hasn't made it into a released version yet.
+* [Chrome](https://bugs.chromium.org/p/chromium/issues/detail?id=750599).
+* [Safari](https://bugs.webkit.org/show_bug.cgi?id=174980).
+
+### In a service worker
+
+I need to finish [the spec for the service worker
+parts](https://github.com/w3c/ServiceWorker/pull/1178), but here's the plan:
+
+As I mentioned before, every `Request` object has a `signal` property. Within a service worker,
+`fetchEvent.request.signal` will signal abort if the page is no longer interested in the response.
+As a result, code like this just works:
+
+    addEventListener('fetch', event => {
+      event.respondWith(fetch(event.request));
+    });
+
+If the page aborts the fetch, `fetchEvent.request.signal` signals abort, so the fetch within the
+service worker also aborts.
+
+If you're fetching something other than `event.request`, you'll need to pass the signal to your
+custom fetch(es).
+
+    addEventListener('fetch', event => {
+      const url = new URL(event.request.url);
+
+      if (event.request.method == 'GET' && url.pathname == '/about/') {
+        // Modify the URL
+        url.searchParams.set('from-service-worker', 'true');
+        // Fetch, but pass the signal through
+        event.respondWith(
+          fetch(url, { signal: event.request.signal })
+        );
+      }
+    });
+
+Follow [the spec](https://github.com/w3c/ServiceWorker/pull/1178) to track this – I'll add links to
+browser tickets once it's ready for implementation.
+
+## The history
+
+Yeah… it took a long time for this relatively simple API to come together. Here's why:
+
+### API disagreement
+
+As you can see [the GitHub discussion is pretty long](https://github.com/whatwg/fetch/issues/27).
+There a lot of nuance in that thread (and some lack-of-nuance), but the key disagreement is one
+group wanted the `abort` method to exist on the object returned by `fetch()`, whereas the other
+wanted a separation between getting the response and effecting the response.
+
+These requirements are incompatible, so one group wasn't going to get what they wanted. If that's
+you, sorry! If it makes you feel better, I was also in that group. But seeing `AbortSignal` fit the
+requirements of other APIs makes it seem like the right choice. Also, allowing chained promises to
+become abortable would become very complicated, if not impossible.
+
+### False starts in TC39
```

Comment:
@domenic is this a fair (but brief) representation of the TC39 stuff? Don't worry about it if you're not keen on revisiting it.